### PR TITLE
Add "exclude_paths" parameter to UberFilesizeTestEngine.

### PR DESCRIPTION
**Pull request by shatlykuber**
Any file which matches at least one given pattern is skipped from
filesize check.

### CODE REVIEW

File: src/unit/engine/UberFilesizeTestEngine.php [hunk 1]
```diff
@@ -57,7 +60,14 @@ private function checkFilesizes() {
         for ($i = 0; $i < count($files); $i++) {
             // deleted files are skipped
             if (file_exists($files[$i])) {
-                if (filesize($files[$i]) > $max_filesize_limit) {
+                $should_be_skipped = False;
+                for ($j = 0; $j < count($exclude_paths); $j++) {
```
Comment (rmaz) [inline src/unit/engine/UberFilesizeTestEngine.php]: Could move this to a function

File: src/unit/engine/UberFilesizeTestEngine.php [hunk 2]
```diff
@@ -66,11 +76,14 @@ private function checkFilesizes() {
         if (count($large_files) > 0) {
             $result->setResult(ArcanistUnitTestResult::RESULT_FAIL);
             $error_string =
-                sprintf('Maximum allowed filesize is %d.',
-                    $max_filesize_limit);
-            $error_string .=  ' Files exceeding the limit:';
+                sprintf('Maximum allowed filesize is %s (%d bytes).',
```
Comment (rmaz) [inline src/unit/engine/UberFilesizeTestEngine.php]: Scrap the `(%d bytes)`

File: src/unit/engine/UberFilesizeTestEngine.php [hunk 3]
```diff
@@ -66,11 +76,14 @@ private function checkFilesizes() {
...
+                    $this->formatBytes($max_filesize_limit), $max_filesize_limit);
+            $error_string .=  ' Files exceeding the filesize limit:';
             for ($i = 0; $i < count($large_files); $i++) {
                 $error_string .= "\n";
+                $large_file_size = filesize($large_files[$i]);
+                $error_string .= $this->formatBytes($large_file_size);
+                $error_string .= ' (' . $large_file_size . ' bytes) ';
```
Comment (rmaz) [inline src/unit/engine/UberFilesizeTestEngine.php]: Make these formatted too

File: src/unit/engine/UberFilesizeTestEngine.php [hunk 4]
```diff
@@ -22,6 +24,7 @@
 final class UberFilesizeTestEngine extends ArcanistUnitTestEngine {
 
     const MAX_FILESIZE_LIMIT_KEY = 'max_filesize_limit';
+    const EXCLUDE_PATHS = 'exlude_paths';
```
Comment (Pawka) [inline src/unit/engine/UberFilesizeTestEngine.php]: `s/exlude/exclude/`

File: src/unit/engine/UberFilesizeTestEngine.php [hunk 5]
```diff
@@ -1,17 +1,19 @@
...
 * { ...
 *   "unit.engine": "UberMultiTestEngine",
 *   "unit.engine.multi.engines":
 *   [
 *     ...,
 *     {
 *       "engine": "UberFilesizeTestEngine",
-*       "max_filesize_limit" : 1000000
+*       "max_filesize_limit" : 1000000,
+*       "exlude_paths": ["(\\.h$)", "(\\.c$)"]
```
Comment (Pawka) [inline src/unit/engine/UberFilesizeTestEngine.php]: Typo here as well.

**Pull request merged**